### ISSUE TITLE
Routing broken in production

### ISSUE THREAD

**Issue by andrewbaisden** [zeit/next.js#10830]
# Bug report

## Describe the bug

When i deploy my Next.js app to Netlify the routing only works when i click on a button that links to the home page. None of the other links for any of the other pages work. I have to manually type the URL into the search bar to get the pages to load. This problem only occurs in production in development the routing works as you would expect when you click on a link.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

https://github.com/andrewbaisden/andrew-baisden-portfolio-2020

Styled components inside a Nav.js component

```jsx
	<MainNav>
		<Link href="/">
			<LinkNav>Home</LinkNav>
		</Link>
			<Link href="/about">
		<LinkNav>About</LinkNav>
			</Link>
		<Link href="/portfolio">
		<LinkNav>Portfolio</LinkNav>
			</Link>
		<Link href="/uses">
		<LinkNav>Uses</LinkNav>
			</Link>
		<Link href="/connect">
		<LinkNav>Connect</LinkNav>
			</Link>
	</MainNav>
```

I have a next.config.js file i commented out the routes because I am not sure that was the problem or if it is even working because it stopped the app from compiling. The experimental code was a solution for for another bug which seems similar to the one i have here. Although i am using styled components and not importing external css stylesheets.

```javascript
// module.exports = {
//     exportPathMap: function() {
//         return {
//             '/': { page: '/' },
//             '/about': { page: '/about' },
//             '/portfolio': { page: '/portfolio' },
//             '/uses': { page: '/uses' },
//             '/connect': { page: '/connect' }
//         };
//     }
// };
module.exports = { experimental: { css: true } };
```

## Expected behavior

The navigation working in production when you click on links which should navigate between pages. External links work fine so it is possible to go to external websites. However internal links don't do anything.

## System information

- OS: macOS
- Browser chrome, safari, firefox
- Version of Next.js: ^9.2.2

## Additional context

N/A

**Comment by timneutkens** [comment]
No reproduction is provided so it'll be hard to look into this.

Did you try hosting on https://zeit.co? https://nextjs.org/docs/deployment#zeit-now-recommended

**Comment by andrewbaisden** [comment]
> No reproduction is provided so it'll be hard to look into this.
> 
> Did you try hosting on https://zeit.co? https://nextjs.org/docs/deployment#zeit-now-recommended

Hey i just updated it with a live URL that i am testing at the moment. My domain is on Netlify so that is where i will deploy it. I will put the backend on Zeit though.

**Comment by rauchg** [comment]
@andrewbaisden try importing the frontend repo and let us know how it goes!

**Comment by andrewbaisden** [comment]
> @andrewbaisden try importing the frontend repo and let us know how it goes!

The frontend repo is imported from github already. I managed to get the links working but i'm not 100% sure how. I imported an empty CSS file i have seen this fix mentioned in issues on here https://github.com/zeit/next.js/blob/master/errors/css-global.md However at the moment my builds are failing to deploy so i'm yet to verify if this fix is the reason why its working.

**Comment by andrewbaisden** [comment]
I finally got it working after more than a day of troubleshooting and I figured out what the problem was. It seems to be some sort of git bug maybe some of the files are getting cached somewhere on git or on Netllify which is corrupting it i’m not sure. The only way i managed to it working was to create a brand new git repo and then deploy it to a new site from git on Netlify.

1. Copy the project into a new folder in preparation for setting up a new repo and delete the git file with its history
2. CD into the folder which has the next app (frontend in my case) and delete the node_modules, .next, next.config.js and package-lock.json file
3. Make sure that all of the components in the pages folder are lower case and not camelcase
4. run `npm install`
5. Setup the new project for git and then deploy it to a NEW git repo. For whatever reason my existing repo fails to get the routing working despite multiple tests and duplicate projects being identical. The only difference was that the new repos i tested only had a master branch whereas my main repo also had a develop branch. And there were only a few commits whereas my main repo had over 80 commits.
6. Deploy to Netlify as a new site from git
7. As my project was in s subfolder called frontend I had to set the base directory to frontend and then run the commands `next build && next export`
